Curses build warning on macOS

# Bug report

### Bug description:

While building on macOS I noticed this warning, seems like a false positive specific for clang on macOS:

```
./Modules/_cursesmodule.c:3141:9: warning: code will never be executed [-Wunreachable-code]
 3141 |         curses_window_set_error(self, "wgetbkgrnd", "getbkgrnd");
      |         ^~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

macOS

<!-- gh-linked-prs -->
### Linked PRs
* gh-152814
<!-- /gh-linked-prs -->

--- a/Modules/_cursesmodule.c
+++ b/Modules/_cursesmodule.c
@@ -3137,7 +3137,8 @@ _curses_window_getbkgrnd_impl(PyCursesWindowObject *self)
     curses_cell_t wcval = {0};
     cursesmodule_state *state = get_cursesmodule_state_by_win(self);
 #ifdef HAVE_NCURSESW
-    if (wgetbkgrnd(self->win, &wcval) == ERR) {
+    int rtn = wgetbkgrnd(self->win, &wcval);  /* avoid -Wunreachable-code on macOS */
+    if (rtn == ERR) {
         curses_window_set_error(self, "wgetbkgrnd", "getbkgrnd");
         return NULL;
     }

